lib/facility_constructor.rb

```diff
@@ -0,0 +1,104 @@
+class FacilityConstructor
+
+  def initialize
+    @parse_hash = {
+      "CO" => :co_parse,
+      "NY" => :ny_parse,
+      "MO" => :mo_parse
+    }
+  end
+
+  def create_facilities(api_hash_array)
+    new_facilities = []
+    api_hash_array.each do |facility_data|
+      facility = Facility.new(parse(facility_data))
+      facility.services = add_services(facility_data)
+      new_facilities << facility
+    end
+    new_facilities
+  end
+
+  def parse(api_hash)
+    self.send(@parse_hash[api_hash[:state]], api_hash)
+    # if api_hash[:state] == "CO"
+    #   parsed_hash = co_parse(api_hash)
+    # elsif api_hash[:state] == "NY"
+    #   parsed_hash = ny_parse(api_hash)
+    # elsif api_hash[:state] == "MO"
+    #   parsed_hash = mo_parse(api_hash)
+    # else
+    #   parsed_hash = api_hash
+    # end
+    # parsed_hash
+  end
+  
+  def co_parse(api_hash)
+    api_hash[:name] = api_hash[:dmv_office]
+    api_hash[:address] = "#{api_hash[:address_li]} #{api_hash[:address__1]} #{api_hash[:city]} #{api_hash[:state]} #{api_hash[:zip]}"
+    api_hash
+  end
+
+  def ny_parse(api_hash)
+    api_hash[:name] = "#{api_hash[:office_name]} #{:office_type}"
+    api_hash[:address] = "#{api_hash[:street_address_line_1]} #{api_hash[:street_address_line_2]} #{api_hash[:city]} #{api_hash[:state]} #{api_hash[:zip_code]}"
+    api_hash[:hours] = ny_hours(api_hash)
+    if api_hash[:public_phone_number] != nil
+      api_hash[:phone] = api_hash[:public_phone_number]
+    else
+      api_hash[:phone] = "(800) 698-2931"
+    end
+    api_hash
+  end
+
+  def ny_hours(api_hash)
+    hours_string = ""
+    if api_hash[:monday_beginning_hours] != nil
```

Comment:
I noticed that every if statement in this method uses _almost_ the same condition, with the days of the week changing. I think you could refactor this with string interpolation and iteration.
```
def ny_hours(api_hash)
  hours_string = ""
  weekdays = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
  weekdays.each do |weekday|
    beginning_hours = api_hash["#{weekday.downcase}_beginning_hours"]
    ending_hours = api_hash["#{weekday.downcase}_ending_hours"]
    if beginning_hours != nil
      hours_string += "#{weekday} - #{beginning_hours}-#{ending_hours}"
    end
  end
  hours_string
end
```